UniversityApp.BussinessLogic.Tests/Services/StudentsServiceTests.cs

```diff
@@ -0,0 +1,259 @@
+            mapper = mapperMockConfig.CreateMapper();
+        }
+
+        [Test]
+        public async Task GetStudent_Should_ReturnsNull_When_StudentDoesntExist([Random(1)] int id)
+        {
+            //Arrange
+            studentsRepositoryMock
+                .GetStudent(Arg.Any<int>())
+                .ReturnsNull();
+            
+            var studentsService = new StudentsService(studentsRepositoryMock, gradesRepositoryMock, mapper);
+
+            //Act
+            var result = await studentsService.GetStudent(id);
+
+            //Assert
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public async Task GetStudent_Should_ReturnsExpectedStudent_WhenStudentExists([Random(1)] int id)
+        {
+            //Arrange
+            var expectedStudent = new Student
+            {
+                FirstName = "Adam",
+                LastName = "Ptak",
+                Age = 24,
+                IsAdult = true,
+                Gender = "M"
+            };
+
+            studentsRepositoryMock.GetStudent(Arg.Any<int>()).Returns(expectedStudent);
+
+            var studentsService = new StudentsService(studentsRepositoryMock, gradesRepositoryMock, mapper);
+
+            //Act
+            var result = await studentsService.GetStudent(id);
+
+            //Assert
+            result.Should().NotBeNull();
+            result.Should().BeEquivalentTo(expectedStudent, options => options.ComparingByValue<StudentDomainModel>().ExcludingMissingMembers());
+        }
+
+        [Test]
+        public async Task GetStudents_Should_ReturnsExpectedStudents_When_Called()
+        {
+            //Arrange
+            var expectedStudents = new List<Student>
+            {
+                new Student
+                {
+                    Id = 22,
+                    FirstName = "Adam",
+                    LastName = "Ptak",
+                    Age = 24,
+                    IsAdult = true,
+                    Gender = "M",
+                    Grades = new List<Grade>()
+                },
+                new Student
+                {
+                    Id = 14,
+                    FirstName = "Lola",
+                    LastName = "Tak",
+                    Age = 23,
+                    IsAdult = true,
+                    Gender = "K",
+                    Grades = new List<Grade>()
+                }
+            };
+
+            studentsRepositoryMock.GetStudents().Returns(expectedStudents);
+
+            var studentsService = new StudentsService(studentsRepositoryMock, gradesRepositoryMock, mapper);
+
+            //Act
+            var result = await studentsService.GetStudents();
+
+            //Assert
+            result.Should().BeEquivalentTo(expectedStudents, options => options.ComparingByValue<StudentDomainModel>().ExcludingMissingMembers());
+        }
+
+        [Test]
+        public async Task SearchStudents_Should_ReturnTupleOfPagedRecodsAndTotalRecordCount_When_Called()
```

Comment:
Name of tests should reflect bussiness logic. No technical data of returned type